https://github.com/NixOS/nixpkgs/actions/runs/12202942991/job/34044818593?pr=362498

Looks like other PRs have same errors, treewide CI trouble?

```
building '/nix/store/sdrvzjsnzr200ygf93y9wffjrnxk438r-attrpaths-superset.json.drv'...
error:
       … from call site
         at /nix/store/8hsambh0xsjzlpmys60fm8lp6irvyi3h-source/pkgs/top-level/release-attrpaths-superset.nix:183:10:
          182|       [ "pkgsMusl" "stdenv" ]
          183|     ] ++ justAttrNames [] filtered;
             |          ^
          184|

       … while calling 'justAttrNames'
         at /nix/store/8hsambh0xsjzlpmys60fm8lp6irvyi3h-source/pkgs/top-level/release-attrpaths-superset.nix:124:25:
          123|   #
          124|   justAttrNames = path: value:
             |                         ^
          125|     let

       … while evaluating a branch condition
         at /nix/store/8hsambh0xsjzlpmys60fm8lp6irvyi3h-source/pkgs/top-level/release-attrpaths-superset.nix:155:9:
          154|       result =
          155|         if tried.success
             |         ^
          156|         then tried.value

       … while calling the 'tryEval' builtin
         at /nix/store/8hsambh0xsjzlpmys60fm8lp6irvyi3h-source/pkgs/top-level/release-attrpaths-superset.nix:152:15:
          151|       seq = builtins.deepSeq attempt attempt;
          152|       tried = builtins.tryEval seq;
             |               ^
          153|

       … while calling the 'deepSeq' builtin
         at /nix/store/8hsambh0xsjzlpmys60fm8lp6irvyi3h-source/pkgs/top-level/release-attrpaths-superset.nix:151:13:
          150|
          151|       seq = builtins.deepSeq attempt attempt;
             |             ^
          152|       tried = builtins.tryEval seq;

       … while calling the 'foldl'' builtin
         at /nix/store/8hsambh0xsjzlpmys60fm8lp6irvyi3h-source/pkgs/top-level/release-attrpaths-superset.nix:142:14:
          141|         else if (value.__attrsFailEvaluation or false) then []
          142|         else lib.pipe value [
         6064|   gcc12Stdenv = overrideCC gccStdenv buildPackages.gcc12;
         6065|   gcc13Stdenv = overrideCC gccStdenv buildPackages.gcc13;
             |   ^
         6066|   gcc14Stdenv = overrideCC gccStdenv buildPackages.gcc14;

       … from call site
         at /nix/store/8hsambh0xsjzlpmys60fm8lp6irvyi3h-source/pkgs/top-level/all-packages.nix:6065:17:
         6064|   gcc12Stdenv = overrideCC gccStdenv buildPackages.gcc12;
         6065|   gcc13Stdenv = overrideCC gccStdenv buildPackages.gcc13;
             |                 ^
         6066|   gcc14Stdenv = overrideCC gccStdenv buildPackages.gcc14;

       … while calling 'overrideCC'
         at /nix/store/8hsambh0xsjzlpmys60fm8lp6irvyi3h-source/pkgs/stdenv/adapters.nix:35:24:
           34|   # Override the compiler in stdenv for specific packages.
           35|   overrideCC = stdenv: cc: stdenv.override {
             |                        ^
           36|     allowedRequisites = null;

       … while evaluating a branch condition
         at /nix/store/8hsambh0xsjzlpmys60fm8lp6irvyi3h-source/pkgs/top-level/all-packages.nix:6051:5:
         6050|   gccStdenv =
         6051|     if stdenv.cc.isGNU
             |     ^
         6052|     then stdenv

       … while evaluating the attribute 'cc.isGNU'
         at /nix/store/8hsambh0xsjzlpmys60fm8lp6irvyi3h-source/pkgs/stdenv/generic/default.nix:168:15:
          [167](https://github.com/NixOS/nixpkgs/actions/runs/12202942991/job/34044818593?pr=362498#step:5:168)|
          [168](https://github.com/NixOS/nixpkgs/actions/runs/12202942991/job/34044818593?pr=362498#step:5:169)|       inherit cc hasCC;
             |               ^
          [169](https://github.com/NixOS/nixpkgs/actions/runs/12202942991/job/34044818593?pr=362498#step:5:170)|

       error: infinite recursion encountered
       at /nix/store/8hsambh0xsjzlpmys60fm8lp6irvyi3h-source/pkgs/top-level/all-packages.nix:6051:8:
         6050|   gccStdenv =
         6051|     if stdenv.cc.isGNU
             |        ^
         6052|     then stdenv
Command exited with non-zero status 1
        Command being timed: "nix-instantiate --eval --strict --json --show-trace /nix/store/8hsambh0xsjzlpmys60fm8lp6irvyi3h-source/pkgs/top-level/release-attrpaths-superset.nix -A paths -I /nix/store/8hsambh0xsjzlpmys60fm8lp6irvyi3h-source --option restrict-eval true --option allow-import-from-derivation false --arg enableWarnings false"
        User time (seconds): 12.96
        System time (seconds): 1.42
        Percent of CPU this job got: 99%
        Elapsed (wall clock) time (h:mm:ss or m:ss): 0:14.46
        Average shared text size (kbytes): 0
        Average unshared data size (kbytes): 0
        Average stack size (kbytes): 0
        Average total size (kbytes): 0
        Maximum resident set size (kbytes): 4952240
        Average resident set size (kbytes): 0
        Major (requiring I/O) page faults: 0
        Minor (reclaiming a frame) page faults: [183](https://github.com/NixOS/nixpkgs/actions/runs/12202942991/job/34044818593?pr=362498#step:5:184)484
        Voluntary context switches: 130
        Involuntary context switches: 72
        Swaps: 0
        File system inputs: 0
        File system outputs: 336
        Socket messages sent: 0
        Socket messages received: 0
        Signals delivered: 0
        Page size (bytes): 4096
        Exit status: 1
error: builder for '/nix/store/sdrvzjsnzr200ygf93y9wffjrnxk438r-attrpaths-superset.json.drv' failed with exit code 1;
       last 25 log lines:
       >          6052|     then stdenv
       > Command exited with non-zero status 1
       >         Command being timed: "nix-instantiate --eval --strict --json --show-trace /nix/store/8hsambh0xsjzlpmys60fm8lp6irvyi3h-source/pkgs/top-level/release-attrpaths-superset.nix -A paths -I /nix/store/8hsambh0xsjzlpmys60fm8lp6irvyi3h-source --option restrict-eval true --option allow-import-from-derivation false --arg enableWarnings false"
       >  User time (seconds): 12.96
       >     System time (seconds): 1.42
       >    Percent of CPU this job got: 99%
       >       Elapsed (wall clock) time (h:mm:ss or m:ss): 0:14.46
       >   Average shared text size (kbytes): 0
       >   Average unshared data size (kbytes): 0
       >         Average stack size (kbytes): 0
       >         Average total size (kbytes): 0
       >         Maximum resident set size (kbytes): 4[952](https://github.com/NixOS/nixpkgs/actions/runs/12202942991/job/34044818593?pr=362498#step:5:953)240
       >    Average resident set size (kbytes): 0
       >  Major (requiring I/O) page faults: 0
       >   Minor (reclaiming a frame) page faults: 183484
       >         Voluntary context switches: 130
       >        Involuntary context switches: 72
       >       Swaps: 0
       >       File system inputs: 0
       >  File system outputs: 336
       >       Socket messages sent: 0
       >        Socket messages received: 0
       >    Signals delivered: 0
       >   Page size (bytes): 4096
       >        Exit status: 1
       For full logs, run 'nix log /nix/store/sdrvzjsnzr200ygf93y9wffjrnxk438r-attrpaths-superset.json.drv'.
Error: Process completed with exit code 1.
```

--- a/pkgs/by-name/dp/dprint/package.nix
+++ b/pkgs/by-name/dp/dprint/package.nix
@@ -1,7 +1,7 @@
 {
   lib,
   stdenv,
-  fetchCrate,
+  fetchFromGitHub,
   rustPlatform,
   installShellFiles,
   testers,
@@ -13,19 +13,28 @@ rustPlatform.buildRustPackage rec {
   pname = "dprint";
   version = "0.47.6";
 
-  src = fetchCrate {
-    inherit pname version;
-    hash = "sha256-7tGzSFp7Dnu27L65mqFd7hzeFFDfe1xJ6cMul3hGyJs=";
+  # Prefer repository rather than crate here
+  #   - They have Cargo.lock in the repository
+  #   - They have WASM files in the repository which will be used in checkPhase
+  src = fetchFromGitHub {
+    owner = "dprint";
+    repo = "dprint";
+    rev = "refs/tags/${version}";
+    hash = "sha256-zyiBFZbetKx0H47MAU4JGauAmthcuEdJMl93M6MobD8=";
   };
 
-  cargoHash = "sha256-y3tV3X7YMOUGBn2hCmxsUUc9QQleKEioTIw7SGoBvSQ=";
-
-  # Tests fail because they expect a test WASM plugin. Tests already run for
-  # every commit upstream on GitHub Actions
-  doCheck = false;
+  cargoHash = "sha256-XuzxoJgJJl4Blw1lDnCG3faEqL9U40MhZEb9LYjiaSs=";
 
   nativeBuildInputs = lib.optionals (stdenv.buildPlatform.canExecute stdenv.hostPlatform) [
     installShellFiles
+  ];
+
+  checkFlags = [
+    # Require creating directory and network access
+    "--skip=plugins::cache_fs_locks::test"
+    "--skip=utils::lax_single_process_fs_flag::test"
+    # Require cargo is running
+    "--skip=utils::process::test"
   ];
 
   postInstall = lib.optionalString (stdenv.buildPlatform.canExecute stdenv.hostPlatform) ''

--- a/pkgs/by-name/dp/dprint/package.nix
+++ b/pkgs/by-name/dp/dprint/package.nix
@@ -59,7 +59,10 @@ rustPlatform.buildRustPackage rec {
     changelog = "https://github.com/dprint/dprint/releases/tag/${version}";
     homepage = "https://dprint.dev";
     license = licenses.mit;
-    maintainers = with maintainers; [ khushraj ];
+    maintainers = with maintainers; [
+      khushraj
+      kachick
+    ];
     mainProgram = "dprint";
   };
 }

